### PR TITLE
Add Generic queue and implementations

### PR DESCRIPTION
**Pull request by ngenovese11**
Adds a generic queue and string/byte implementations.  Also adds a class that processes string responses from a IBasicCommunication or Gather

### CODE REVIEW

File: essentials-framework/Essentials Core/PepperDashEssentialsBase/Queues/GenericQueue.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+using Crestron.SimplSharpPro.CrestronThread;
+using PepperDash.Core;
+
+namespace PepperDash_Essentials_Core.Queues
+{
+    /// <summary>
+    /// Threadsafe processing of queued items with pacing if required
+    /// </summary>
+    /// <typeparam name="T">Type of item to be processed</typeparam>
+    public class GenericQueue<T> : IQueue<T> where T : class
```
Comment (andrew-welker) [inline essentials-framework/Essentials Core/PepperDashEssentialsBase/Queues/GenericQueue.cs:12]: I'm thinking that we may want to create an IQueueMessage interface, then restrict messages here to implement that interface. That makes it easier to create the action to process the queue too. Thoughts?

Comment (ngenovese11) [reply essentials-framework/Essentials Core/PepperDashEssentialsBase/Queues/GenericQueue.cs:12]: Sure something like the below pseudo code-

```
public interface IQueueMessage
{
   void Dispatch()
}
```

and then the queue would just invoke the dispatch method?

Comment (ngenovese11) [reply essentials-framework/Essentials Core/PepperDashEssentialsBase/Queues/GenericQueue.cs:12]: Then you could have something like 

```
public class MessageDispatch : IQueueMessage
{
  private readonly IBasicCommunication _coms;
  private readonly string _message;

  public MessageDispatch(IBasicCommunication coms, string message)
  {
    _coms = coms
   _message = message;
  }

public void Dispatch()
{
   _coms.SendText(_message);
}
```

Comment (andrew-welker) [reply essentials-framework/Essentials Core/PepperDashEssentialsBase/Queues/GenericQueue.cs:12]: Exactly.

Comment (ngenovese11) [reply essentials-framework/Essentials Core/PepperDashEssentialsBase/Queues/GenericQueue.cs:12]: Sounds good, let me work it up

Comment (andrew-welker) [reply essentials-framework/Essentials Core/PepperDashEssentialsBase/Queues/GenericQueue.cs:12]: Then on the `GenericQueue` class you can do `public class GenericQueue<T> where T:IQueueMessage`

File: essentials-framework/Essentials Core/PepperDashEssentialsBase/Queues/GenericQueue.cs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        /// <summary>
+        /// If the instance has been disposed.
+        /// </summary>
+        public bool Disposed { get; private set; }
+
+        /// <summary>
+        /// Constructor for generic queue with no pacing
+        /// </summary>
+        /// <param name="key">Key</param>
+        public GenericQueue(string key)
```
Comment (andrew-welker) [inline essentials-framework/Essentials Core/PepperDashEssentialsBase/Queues/GenericQueue.cs:30]: We should add another constructor with an option for queue size. Crestron Queues default to 5, and don't automatically increase their size

**Review by andrew-welker — APPROVED** [review]

**Pull request merged**